File: abcd_fl/lib/controllers/models/auth_model.dart
class PatientModel {
  String? id;
  String? patientCode;
  String? firstName;
  String? lastName;
  String? email;
  String? mobile;
  bool? isAppEnabled;

  PatientModel({
    this.id,
    this.patientCode,
    this.firstName,
    this.lastName,
    this.email,
    this.mobile,
    this.isAppEnabled,
  });

  factory PatientModel.fromJson(Map<String, dynamic> json) {
    return PatientModel(
      id: json['id'],
      patientCode: json['patientCode'],
      firstName: json['firstName'],
      lastName: json['lastName'],
      email: json['email'],
      mobile: json['mobile'],
      isAppEnabled: json['isAppEnabled'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'patientCode': patientCode,
      'firstName': firstName,
      'lastName': lastName,
      'email': email,
      'mobile': mobile,
      'isAppEnabled': isAppEnabled,
    };
  }
}

class AuthResponseDataModel {
  String? accessToken;
  String? refreshToken;
  String? tokenType;
  int? expiresIn;
  int? refreshExpiresIn;
  PatientModel? patient;

  AuthResponseDataModel({
    this.accessToken,
    this.refreshToken,
    this.tokenType,
    this.expiresIn,
    this.refreshExpiresIn,
    this.patient,
  });

  factory AuthResponseDataModel.fromJson(Map<String, dynamic> json) {
    return AuthResponseDataModel(
      accessToken: json['accessToken'] ?? json['token'],
      refreshToken: json['refreshToken'],
      tokenType: json['tokenType'],
      expiresIn: json['expiresIn'],
      refreshExpiresIn: json['refreshExpiresIn'],
      patient: json['patient'] != null
          ? PatientModel.fromJson(json['patient'])
          : (json['user'] != null ? PatientModel.fromJson(json['user']) : null),
    );
  }
}

class BaseResponseModel {
  bool? success;
  String? message;

  BaseResponseModel({this.success, this.message});

  factory BaseResponseModel.fromJson(Map<String, dynamic> json) {
    return BaseResponseModel(
      success: json['success'],
      message: json['message'],
    );
  }
}

File: abcd_rn/src/app/redux/models/authModel.ts
export interface LoginBodyActionType {
  email?: string;
  password?: string;
  deviceIdentifier?: string;
  deviceName?: string;
  deviceType?: string;
}

export interface AuthLoginResponseModel {
  success?: boolean;
  message?: string;
  data?: {
    accessToken?: string;
    refreshToken?: string;
    tokenType?: string;
    expiresIn?: number;
  } | null;
}

export interface RefreshResponseModel {
  success?: boolean;
  message?: string;
  data?: {
    accessToken?: string;
    refreshToken?: string;
    tokenType?: string;
    expiresIn?: number;
  } | null;
}

// {
//   "success": false,
//   "statusCode": 400,
//   "message": "Foreign key constraint failed",
//   "timestamp": "2026-07-25T07:26:09.447Z",
//   "path": "/api/v1/auth/register"
// }

export interface RegisterBodyActionType {
  name?: string;
  lastName?: string;
  email?: string;
  phone?: string;
  password?: string;
}

export interface RegisterResponseModel {
  success?: boolean;
  message?: string;
  data?: {
    sent?: boolean;
    email?: string;
    message?: string;
  } | null;
}

export interface VerifyRegisterBodyActionType {
  email?: string;
  code?: string;
}

export interface VerifyRegisterResponseModel {
  success?: boolean;
  message?: string;
  data?: {
    accessToken?: string;
    refreshToken?: string;
    tokenType?: string;
    expiresIn?: number;
    refreshExpiresIn?: number;
    patient?: {
      id?: string;
      patientCode?: string;
      firstName?: string;
      lastName?: string;
      email?: string;
      mobile?: string;
      isAppEnabled?: boolean;
    };
  } | null;
}

export interface ResendRegisterOtpBodyActionType {
  email?: string;
}

export interface ResendRegisterOtpResponseModel {
  success?: boolean;
  message?: string;
  data?: {
    sent?: boolean;
    email?: string;
    message?: string;
  } | null;
}

export interface ForgotPasswordBodyActionType {
  email: string;
}

export interface ForgotPasswordResponseModel {
  success?: boolean;
  message?: string;
  data?: {
    sent?: boolean;
  } | null;
}

export interface ResetPasswordBodyActionType {
  email: string;
  code: string;
  newPassword?: string;
}

export interface ResetPasswordResponseModel {
  success?: boolean;
  message?: string;
  data?: {
    reset?: boolean;
  } | null;
}
